streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -236,18 +243,30 @@ public ProcessorNode build() {
         private boolean isMatch(String topic) {
             return this.pattern.matcher(topic).matches();
         }
+
+        @Override
+        TopologyDescription.Source describe() {
```

Comment:
It's all the same. There is no inconsistency IMHO. All three constructors are just setting members and we call a `new` for all three in the corresponding `NodeFactory` -- the only different is, that for source we need to distinguish between pattern vs topic subscription and need a new lines of code to set up the constructor argument (this step is not required for processors and sinks)